tests/features/bootstrap/FeatureContext.php

```diff
@@ -212,4 +212,42 @@ public function createLanguages($langcode) {
     $this->languageCreate((object) ['langcode' => $langcode]);
   }
 
+  /**
+   * Prepare for PHP errors log.
+   *
+   * @BeforeSuite
+   */
+  public static function preparePhpErrors($event) {
+    // Clear out the watchdog table at the beginning of each test suite.
+    db_truncate('watchdog')->execute();
+  }
+
+  /**
+   * Check for PHP errors log.
+   *
+   * @AfterSuite
+   */
+  public static function checkPhpErrors($event) {
+    // Find any PHP errors at the end of the suite
+    // and output them as an exception.
+    $log = db_select('watchdog', 'w')
+      ->fields('w')
+      ->condition('w.type', 'php', '=')
+      ->execute()
+      ->fetchAll();
+    if (!empty($log)) {
+      $errors = count($log);
+      $message = "$errors PHP errors were logged to the watchdog in this suite:\n\n";
+      foreach ($log as $error) {
+        $error->variables = unserialize($error->variables);
+        $date = date('Y-m-d H:i:sP', $error->timestamp);
+        $message .= sprintf("Message: %s: %s in %s (line %s of %s).\n", $error->variables['%type'], $error->variables['!message'], $error->variables['%function'], $error->variables['%line'], $error->variables['%file']);
+        $message .= "Location: $error->location\n";
+        $message .= "Referer: $error->referer\n";
+        $message .= "Date/Time: $date\n\n";
+      }
+      throw new Exception($message);
+    }
```

Comment:
It would be great to include some information about the actual test that caused the bug, like the file where the error occurred, and the line number of the step where the error occurred.

You can get the filename from `$event->getFeature()->getFile()`, the line number from `$event->getStep()->getLine()`. The content of the step definition is in `$event->getStep()->getText()`